### PR TITLE
Update quote preview items

### PR DESCRIPTION
**Pull request by lexerom**
## Summary

[Preview](https://preview.redoc.ly/all-rebilly/update-quote-preview-items/tag/Quotes/operation/PostQuote/#!c=201&path=invoicePreview&t=response)

Add initialAmounts and recurringAmounts subsets instead of a one amounts subset.
Add setupUnitPrice if an item has setup price.
Add trialUnitPrice and trialPeriod if an item has trial.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by tamasvigh — DISMISSED** [review]

File: openapi/components/schemas/Quote.yaml [hunk 1]
```diff
@@ -109,38 +109,75 @@ properties:
...
+            description: Tax amount that is applied to the invoice.
+            type: number
+            x-type: Money
+            format: double
+      recurringAmounts:
+        type: object
+        description: |-
+          Total amounts of the recurring invoice.
+          This includes recurring items only.
+          Can be `null` if the quote doesn't have recurring items.
```
Comment (noon-dawg) [inline openapi/components/schemas/Quote.yaml:153]: suggestion: 
```suggestion
          If the quote does not have recurring items, the value of this field is `null`.
```

File: openapi/components/schemas/Quote.yaml [hunk 2]
```diff
@@ -176,27 +213,43 @@ properties:
...
+              type: number
+              format: double
+              nullable: true
+            setupTaxAmount:
+              description: Invoice setup item tax.
+              type: number
+              format: double
+              nullable: true
+            trialTaxAmount:
+              description: Invoice trial item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/Quote.yaml:249]: nitpick: [noun stack](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#noun-strings-or-noun-stacks)

I'm not sure what these field represent. I made an effort to expand on them, not sure if they are correct.

```suggestion
              description: Tax amount of the setup that is displayed on the invoice.
              type: number
              format: double
              nullable: true
            trialTaxAmount:
              description: Tax amount of the trial that is displayed on the invoice.
```

File: openapi/components/schemas/StorefrontQuote.yaml [hunk 3]
```diff
@@ -94,38 +94,75 @@ properties:
...
+            description: Tax amount that is applied to the invoice.
+            type: number
+            x-type: Money
+            format: double
+      recurringAmounts:
+        type: object
+        description: |-
+          Total amounts of the recurring invoice.
+          This includes recurring items only.
+          Can be `null` if the quote doesn't have recurring items.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuote.yaml:138]: suggestion: if possible, it's good to describe scenario before the result
```suggestion
          If the quote does not have recurring items, the value of this field is `null`.
```

**Review by tamasvigh — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

File: openapi/components/schemas/Quote.yaml [hunk 4]
```diff
@@ -109,38 +109,75 @@ properties:
...
-      taxAmount:
-        description: Tax amount that is applied to the invoice.
-        type: number
-        x-type: Money
-        format: double
+      initialAmounts:
+        type: object
+        description: Total amounts of the initial invoice.
+        properties:
+          amount:
```
Comment (ikram-patah) [inline openapi/components/schemas/Quote.yaml:123]: quick question: How is `amount` different from `subtotalAmount`?

**Review by ikram-patah — APPROVED** [review]

File: openapi/components/schemas/Quote.yaml (hunk 4 above)
Comment (lexerom) [reply openapi/components/schemas/Quote.yaml:123]: subtotalAmount is just sum of items without discounts, shipping and taxes.

**Review by i-yasko — APPROVED** [review]

**Pull request merged**